lib/access-context.js

```diff
@@ -126,9 +130,29 @@ AccessContext.prototype.addPrincipal = function(principalType, principalId, prin
 AccessContext.prototype.getUserId = function() {
   for (var i = 0; i < this.principals.length; i++) {
     var p = this.principals[i];
+    if (p.type === Principal.APP || p.type === Principal.ROLE || p.type == Principal.SCOPE) {
+      continue;
+    }
+    // the principalType must either be 'USER'
     if (p.type === Principal.USER) {
       return p.id;
     }
+    // or permit to resolve a valid user model
+    var userModel = this.registry.findModel(p.type);
+    if (!userModel) continue;
+    if (userModel.modelName === 'User') {
+      return p.id;
+    }
+    var modelBase = userModel.definition.settings.base;
+    while (modelBase) {
+      // make sure modelBase is a model instance
+      modelBase = typeof(modelBase === 'string') ? this.registry.findModel(modelBase) : modelBase;
+      if (modelBase.modelName === 'User') {
+        return p.id;
+      }
+      // iterate on the upper base model
+      modelBase = modelBase.definition.settings.base;
+    }
```

Comment:
The way i get `Registry.getModel()`: `this.registry.getModel('User')` would actually return **one among** several models inheriting from built-in User model, not the built-in User Model itself

then i think `userModel.prototype instanceof BaseUser` would only return `true` for models being instances or inheriting from that particular User model

do you see things different?